### PR TITLE
Fix logging and pipeline failure during init

### PR DESCRIPTION
**Pull request by kowshikn**
*Description of changes:*

Cherry-pick commit of the [fix](https://github.com/opendistro-for-elasticsearch/Data-Prepper/pull/197) applied on the release. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yadavcbala** [comment]
Thought we already merged this on Friday ?

**Comment by yadavcbala** [comment]
> Thought we already merged this on Friday ?
my bad it was in v0.1.x branch

**Review by yadavcbala — APPROVED** [review]

File: data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java [hunk 1]
```diff
@@ -111,12 +113,20 @@ private void buildPipelineFromConfiguration(
...
                 LOG.info("Source of pipeline [{}] requires building of pipeline [{}]", sourcePipelineName,
-                        pipelineNameOptional.get());
+                        connectedPipeline);
                 //Build connected pipeline for the pipeline connector to be available
+                //Building like below sometimes yields multiple runs if the pipeline building fails before sink
+                //creation. except for running the creation again, it will not harm anything - TODO Fix this
                 buildPipelineFromConfiguration(pipelineNameOptional.get(), pipelineConfigurationMap, pipelineMap);
             }
+            if (!pipelineMap.containsKey(connectedPipeline)) {
+                LOG.info("Connected Pipeline [{}] failed to build, Failing building source for [{}]",
```
Comment (erosas) [inline data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java:126]: log.warn at least?

**Review by erosas — APPROVED** [review]

File: data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java (hunk 1 above)
Comment (yadavcbala) [reply data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java:126]: my bad - it should have been LOG.error; Erick will fix it in following PR

**Pull request merged**